internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -480,6 +481,23 @@ func (r *BMACReconciler) reconcileSpokeBMH(ctx context.Context, bmh *bmh_v1alpha
 		return reconcileError{err}
 	}
 
+	// Add detached annotation to hub BMH so that Ironic stops managing the host from
+	// the hub cluster. After spoke BMH is created, the host will be managed by the spoke
+	// cluster.
+	bmhAnnotations := bmh.ObjectMeta.GetAnnotations()
+	if _, ok := bmhAnnotations[BMH_DETACHED_ANNOTATION]; !ok {
+		if bmh.ObjectMeta.Annotations == nil {
+			bmh.ObjectMeta.Annotations = make(map[string]string)
+		}
+		bmh.ObjectMeta.Annotations[BMH_DETACHED_ANNOTATION] = "true"
+
+		err = r.Client.Update(ctx, bmh)
+		if err != nil {
+			r.Log.WithError(err).Errorf("Error adding BMH detached annotation before creating spoke BMH.")
+			return reconcileError{err}
+		}
+	}
+
```

Comment:
To keep things consistent, I'm wondering if we should do this after the `ensureSpokeBMH` call and return `reconcileComplete{true}` so that the actual update call to the original BMH can be done in the main `Reconcile` function.

The big question is whether we want to have the source BMH detached before completing the creation of the spoke BMH or not. A case could be made for either case that one should happen before the other but I think detaching the BMH after the spokeBMH has been created is probably better in this case.

Reply:
Yes we can move it to after `ensureSpokeBMH` and let the original BMH update happen in the main `Reconcile` function.